code-challenges.test.js

```diff
@@ -15,42 +15,236 @@
+//
+//   ● smOdd › takes in an array and returns a new array of only odd numbers sorted from least to greatest.
+//
+//     ReferenceError: smOdd is not defined
 
 // b) Create the function that makes the test pass.
-
+//create a function named smOdd (small odds)
+//it takes in an array for an input
+//it outputs a smaller version of the array with only odd numbers sorted from least to greatest.
+//built in method to .filter with a type of and modulo that doesnt equal 0 to return the required data of only odd numbers, and .sort() to then sort them
+
+//i started with writing in-line because it did not seem that complex to start. i used yarn jest to constantly check the values I was returning:
+// Array [
+//    -   -9,
+//    +   9,
+//    +   "7",
+//    +   true,
+//        7,
+//    -   9,
+//        199,
+//      ]
+//My first result showed me that it was returning strings, and booleans so I knew I had to add a typeof checker to only get numbers.
+// Array [
+//     -   -9,
+//     -   7,
+//         9,
+//     +   7,
+//         199,
+//     +   -9,
+//       ]
+//then i knew I had too reverse the sorting feature from smallest to biggest
+const smOdd = (array) => array.sort((b, a) => b - a).filter(value => typeof value === "number" && value%2 !== 0)
+
+//Got Green - this was kind of fast so I tried refactoring below! What if NaN was included in the array?
+// PASS  ./code-challenges.test.js
+//   fibSeq
+//     ✓ takes in a number (greater than 2) and returns an array that length containing the numbers of the Fibonacci sequence (2 ms)
+//   smOdd
+//     ✓ takes in an array and returns a new array of only odd numbers sorted from least to greatest.
+//
+// Test Suites: 1 passed, 1 total
+//for refactoring maybe I could add something that turns the string numbers into actual numbers? just a thought. A little research showed me parsint() - wonder if it would work on inside the filter.
+
+// const smOdd = (array) => array.sort((b, a) => b - a).map(value => parseInt(value, 10))
+// [
+//    -9,   0, NaN, NaN,   4,
+//     7,   7,   8,   9, NaN,
+//   199, NaN
+// ]
+//this was the best result I could get I know I can then run filter on smOdd but I wanted to get them all in one line if possible.
+//I used parseInt(value, 10) - 10 because radix changes because map has 3 values it passes through (index) which can change the actual radix of the parseInt? I'm not too familiar with it but it was something like how to computer reads numbers
+//well refactored I got there! it is longer but i think more flexible? it converts any strings of numbers into numbers, and gets rid of all NaN by using a filter for any Booleans
+// const smOdd = (array) => {
+//   let sorted = array.sort((b, a) => b - a).map(value => parseInt(value, 10))
+//   sorted = sorted.filter(value => typeof value === "number" && value%2 !== 0)
+//   return sorted.filter(Boolean)
+// }
+//to refactor the last attempt i think it might be possible to combine map, and filter into one line
```

Comment:
Great ork on  all these refactors. Your documentation of your thought process will help immensely as you look back at these problems in the near future.